### PR TITLE
Added an AppSettings option to disable Sqlite connection pooling.

### LINKED ISSUE #11387: Add option to prevent YesSql file locking
I'm currently upgrading a project to .Net 6.0 and realized that I need to modify the configuration. Using the pooled Sqlite settings uses a lock which prevents our backup process from backing up the Sqlite db. I need to add `;Pooling=False` to the connection string, but it doesn't appear that there is an API for this.

On the cheap, even something as being able to configure `DatabaseFileName` would be fine, as we could overwrite `yessql.db` with `yessql.db;Pooling=False` (aka Connection String Injection) and then the connection string would be `Data Source=[etc]/yessql.db;Pooling=False;Cache=Shared`.

Not sure if this would conflict with the `PoolingJsonNetSerializer` or other assumptions that Orchard Core makes about the pooled nature of connections. If so, there should be a more involved way to configure Sql like `IOptions<OrchardCoreSqliteOptions>.Value.IsPooled`

### PR DESCRIPTION
**Pull request by johnrom**
closes #11387 

Sqlite locks the file when connection pooling is enabled, which can prevent certain backup processes, App Service cloning processes, etc from working. This PR adds an option to disable connection pooling and remove the lock from the Sqlite DB when it isn't being written to.

### Testing

- Pull this PR
- Launch .Net Core Launch (web debug)
- Install the site with an Sqlite Multitenant profile.
- Try to delete `App_Data/Sites/Default/yessql.db`
  > ❌ You'll get an error because the file is locked.
- Stop the application
- Open `OrchardCore.Cms.Web/appsettings.json` and uncomment these lines:
  ```json
  "OrchardCore_Data_Sqlite": {
    "UseConnectionPooling": false
  },
  ```
- Launch .Net again
- Delete `App_Data/Sites/Default/yessql.db`
  >  ✓ The file will be deleted.

### CODE REVIEW

**Review by Skrypt — APPROVED** [review]

**Comment by Skrypt** [comment]
Why not. The code is great.

**Review by hishamco — CHANGES_REQUESTED** [review]
LGTM, it would be nice if you added docs into the introduced types

**Comment by ns8482e** [comment]
@johnrom @Skrypt lets also add database file name as part of options

**Comment by hishamco** [comment]
Good point Niraj I already created an old PR, I think it's the time to make configurable

**Comment by ns8482e** [comment]
These properties are also worth (debatable) adding to  `SqliteOptions`

- DatabaseFileName ( with default value to `yessql.db`)
- Max Pool Size
- DateTimeFormat 
- Cached
- CacheSize
- Password

**Comment by hishamco** [comment]
> DatabaseFileName ( with default value to yessql.db)

`OC.db` ;)

> Password (default empty)

I'm not sure if it's really works, I try it long time back with no luck

**Comment by sebastienros** [comment]
At this point with all the options it's just a connection string, so might as well allow for custom connection strings on setup that can also be changed in the settings (like the other databases). And probably provide a default value for sqlite.

**Comment by sebastienros** [comment]
But here it's a global app option to disable pooling, so I'd rather use this PR as-is without other "tenant-specific" options like filename and so on.

**Comment by johnrom** [comment]
@hishamco I added docs to the docs, but neglected the types. I'll try and circle back today or tomorrow if the PR doesn't get shut down in favor of a raw connection string.

@ns8482e I wanted to keep this PR specific to the issue I was trying to solve -- it can serve as a basis for additional configuration in a separate issue / PR.

@sebastienros It could probably be a connection string, but I was considering the fact that OC might change how it works depending on whether certain options are default or not. I personally would rather provide specific options and then build the connection string using ConnectionStringBuilder instead of accepting any old value because:

- Removing Pooling or changing Caching behavior etc, might someday require some other service to act differently.
- A new feature might be released for Sqlite and someone might opt in even though OC would completely break because of it.

**Comment by ns8482e** [comment]
> @ns8482e I wanted to keep this PR specific to the issue I was trying to solve -- it can serve as a basis for additional configuration in a separate issue / PR.

IMHO It's worth adding more options to `SqliteOptions`,  and I also agree it shouldn't be  blocking this PR

File: src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs
Comment (ns8482e) [reply src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs:5]: Nice if we use Has/Is/Can prefix for `bool`

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs:5]: What about `UsePooling`?

Comment (ns8482e) [reply src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs:5]: I am fine - it just `PoolConnections` feels like #s  (list/array kind)

Comment (johnrom) [reply src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs]: Changed to `UseConnectionPooling`.

**Comment by johnrom** [comment]
Switched option name to `UseConnectionPooling` per https://github.com/OrchardCMS/OrchardCore/pull/11393#pullrequestreview-913072831 and added doc comments where necessary per https://github.com/OrchardCMS/OrchardCore/pull/11393#pullrequestreview-912928472.

**Review by ns8482e — APPROVED** [review]

**Review by hishamco — APPROVED** [review]

**Pull request merged**
